aiutils/tftools/plholder_management.py

```diff
@@ -0,0 +1,141 @@
+"""This module defines a helper class for managing placeholders in Tensorflow.
+
+The PlholderManager class allows easy management of placeholders including 
+adding placeholders to a Tensorflow graph, producing easy access to the added
+placeholders using a dictionary with placeholder names as keys, create feed 
+dictionary from a given input dictionary, and print placeholders and feed dict 
+to standard output. 
+
+More importantly, the class allows sparse scipy matrices to 
+be passed into graphs (Tensorflow currently allows only dense matrices to be fed
+into placeholders).
+"""
+import tensorflow as tf
+import numpy as np
+import scipy.sparse as sps
+import pdb
+
+
+class PlholderManager():
+    """Class for managing placeholders."""
+
+    def __init__(self):
+        self._plholders = dict()
+        self.issparse = dict()
+
+    def add_plholder(self, name, dtype, shape=None, sparse=False):
+        """Add placeholder.
+        
+        If the sparse is True then 3 placeholders are automatically created 
+        corresponding to the indices and values of the non-zero entries, and 
+        shape of the sparse matrix. The user does not need to keep track of 
+        these and can directly pass a sparse scipy matrix as input and a 
+        Tensorflow SparseTensor object is made available for use in the graph.
+
+        Args:
+            name (str): Name of the placeholder. 
+            dtype (tf.Dtype): Data type for the placeholder.
+            shape (list of ints): Shape of the placeholder.
+            sparse (bool): Specifies if the placeholder takes sparse inputs.
+        """
+
+        self.issparse[name] = sparse
+        if not sparse:
+            self._plholders[name] = tf.placeholder(dtype, shape, name)
+
+        else:
+            name_indices = name + '_indices'
+            name_values = name + '_values'
+            name_shape = name + '_shape'
+
+            self._plholders[name_indices] = tf.placeholder(tf.int64, [None, 2],
+                                                           name_indices)
+            self._plholders[name_values] = tf.placeholder(dtype, [None],
+                                                          name_values)
+            self._plholders[name_shape] = tf.placeholder(tf.int64, [2],
+                                                         name_shape)
+
+    def get_plholders(self):
```

Comment:
So you are saying - instead of returning a plholders which is a dict, instead have a function that takes plholder name and returns the placeholder? That could be done but then I find it convenient to have a dict of placeholders that can be passed to functions, otherwise the plholder management object would have to be passed.